mmv1/templates/terraform/examples/looker_instance_enterprise_full.tf.erb

```diff
@@ -0,0 +1,90 @@
+resource "google_looker_instance" "<%= ctx[:primary_resource_id] %>" {
+  name               = "<%= ctx[:vars]["instance_name"] %>"
+  platform_edition   = "LOOKER_CORE_ENTERPRISE_ANNUAL"
+  region             = "us-central1"
+  private_ip_enabled = true
+  public_ip_enabled  = true
+  reserved_range     = "${google_compute_global_address.looker_range.name}"
+  consumer_network   = "projects/${data.google_project.project.number}/global/networks/${data.google_compute_network.looker_network.name}"
+  admin_settings {
+    allowed_email_domains = ["google.com"]
+  }
+  encryption_config {
+    kms_key_name = google_kms_crypto_key.looker_crypto_key.id
+  }
+  maintenance_window {
+    day_of_week = "THURSDAY"
+    start_time {
+      hours   = 22
+      minutes = 0
+      seconds = 0
+      nanos   = 0
+    }
+  }
+  deny_maintenance_period {
+    start_date {
+      year = 2050
+      month = 1
+      day = 1
+    }
+    end_date {
+      year = 2050
+      month = 2
+      day = 1
+    }
+    time {
+      hours = 10
+      minutes = 0
+      seconds = 0
+      nanos = 0
+    }
+  }
+  oauth_config {
+    client_id = "<%= ctx[:vars]["client_id"] %>"
+    client_secret = "<%= ctx[:vars]["client_secret"] %>"
+  }
+  depends_on   = [
+    google_service_networking_connection.looker_vpc_connection,
+    google_kms_crypto_key.looker_crypto_key
+  ]
+}
+
+resource "google_service_networking_connection" "looker_vpc_connection" {
+  network                 = data.google_compute_network.looker_network.id
+  service                 = "servicenetworking.googleapis.com"
+  reserved_peering_ranges = [google_compute_global_address.looker_range.name]
+}
+
+resource "google_compute_global_address" "looker_range" {
+  name          = "<%= ctx[:vars]["address_name"] %>"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 20
+  network       = data.google_compute_network.looker_network.id
+}
+
+data "google_project" "project" {}
+
+// This example assumes this network already exists.
+// If this network hasn't been created and you are using this example in your
+// config, add an additional network resource or change
+// this from "data" to "resource"
+data "google_compute_network" "looker_network" {
+  name = "looker-network"
+}
+
+resource "google_kms_key_ring" "looker_key_ring" {
```

Comment:
We may need to bootstrap this one out of band, since some KMS resources can't be deleted. There should be a way to do it within the example, let me get back to you once I've figured it out.

Reply:
Ah, Redis's network was an example of the pattern. See https://github.com/GoogleCloudPlatform/magic-modules/blob/499e9487546f0926a6c364afb509d8bad4fc9a32/mmv1/products/datastream/Stream.yaml#L70-L89 and https://github.com/GoogleCloudPlatform/magic-modules/blob/499e9487546f0926a6c364afb509d8bad4fc9a32/mmv1/templates/terraform/examples/datastream_stream_full.tf.erb#L4 for a reference for how we'll need to plumb this in. We can use a datasource if you want, or just supply the value directly.

Reply:
Specifically `stream_cmek: 'acctest.BootstrapKMSKeyInLocation(t, "us-central1").CryptoKey.Name'` is the relevant bit.

Reply:
Done!